### ISSUE TITLE
Minimalistic (REST ?) API for interfacing (to Python , ...) 

### ISSUE THREAD

**Issue by breznak** [nupic-community/nupic.visualizations#42]
Suggested parts of the API:
- [ ] `init()` graph (start the application, not necessary, we can open the browser manually)
- [ ] `updateData(newBatch)`

What are the right methods to interfacing JS calls? 
- a REST API? 

Use-case:
- I'd like to interface our graph from other language (python). My example usecase is running a HTM model:

```
htm = NuPIC.init()
graph = ... //open browser or somehow start the app

for ... :
  result = htm.compute(input)
  graph.updateData(result) // append new data
```

The API idea follows from my comment https://github.com/nupic-community/nupic.visualizations/issues/40#issuecomment-163790650 

It could overcome the problem with polling updates to a file (impossible by Papa) while `step` updates are possible in DyGraphs, this could replace #17

**Comment by breznak** [comment]
CC @rhyolight would this work for you? 
This python-to-JS must be something you have to solve in your "htm.smartthings..." repo, can you show me what the best approach to interfacing would be? (the REST,...)
@jefffohl This is another approach to #27 - if we could implement the `graph.updateOptions()` as you seggested to update with new data, then this could work (?) And potentially giving better functionality. 
Or ideally both - API updates and updated-file polling.

**Comment by rhyolight** [comment]
I think that the charting codebase here in this repo should be a client-only API for in-browser charting (like Dygraphs). If you want to create a server for relaying data into the charts, it would like be specific to one kind of data or another. I suggest that (the server-side work) be done in other projects and keep this repo client-only.

**Comment by breznak** [comment]
@rhyolight I disagree a bit here. Maybe we could discuss what each of us wants from the repo somewhere? I have a "tool for insight, exploring, presenting (anomaly) (time-series) data" in mind. 
Anyway, maybe this issue should not say an API but a "method to add additional data" to an existing graph, grammatically. I've suggested that for the #17 - streaming data plotting - which I think is a feature that differentiates us from other tools.
